skrub/_make_tabular_pipeline.py

```diff
@@ -0,0 +1,196 @@
+import sklearn
+from sklearn import ensemble
+from sklearn.base import BaseEstimator
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import OrdinalEncoder
+from sklearn.utils.fixes import parse_version
+
+from ._minhash_encoder import MinHashEncoder
+from ._table_vectorizer import TableVectorizer
+from ._to_categorical import ToCategorical
+
+
+def make_tabular_pipeline(predictor, n_jobs=None):
+    """Get a simple machine-learning pipeline that should work well in many cases.
+
+    This function returns a scikit-learn :obj:`~sklearn.pipeline.Pipeline` that
+    combines a :obj:`TableVectorizer`, a :obj:`~sklearn.impute.SimpleImputer`
+    if missing values are not handled by the provided ``predictor``, and
+    finally the ``predictor`` itself.
+
+    This pipeline is simple but (depending on the chosen ``predictor``) should
+    provide a strong baseline for many learning problems. It can handle tabular
+    input and complex data such as categories, text or datetimes.
+
+    Parameters
+    ----------
+    predictor : str or scikit-learn estimator
+        The estimator to use as the final step in the pipeline. Appropriate
+        choices are made for previous step depending on the ``predictor``. Can
+        be the string ``"regressor"`` to use a
+        :obj:`~sklearn.ensemble.HistGradientBoostingRegressor` or
+        ``"classifier"`` to use a
+        :obj:`~sklearn.ensemble.HistGradientBoostingClassifier`.
+
+    n_jobs : int, default=None
+        Number of jobs to run in parallel in the :obj:`TableVectorizer` step.
+        ``None`` means 1 unless in a joblib ``parallel_backend`` context.
+        ``-1`` means using all processors.
+
+    Returns
+    -------
+    Pipeline
+        A scikit-learn :obj:`~sklearn.pipeline.Pipeline` chaining some
+        preprocessing and the provided ``predictor``.
+
+    Examples
+    --------
+    >>> from skrub import make_tabular_pipeline
+
+    We can easily get a default pipeline for classification or regression:
+
+    >>> make_tabular_pipeline('regressor')                # doctest: +SKIP
+    Pipeline(steps=[('tablevectorizer',
+                     TableVectorizer(high_cardinality_transformer=MinHashEncoder(),
+                                     low_cardinality_transformer=ToCategorical())),
+                    ('histgradientboostingregressor',
+                     HistGradientBoostingRegressor(categorical_features='from_dtype'))])
+
+    This pipeline can handle complex, tabular data:
+
+    >>> import pandas as pd
+    >>> X = pd.DataFrame({'a': ['2020-01-02', '2021-04-01'], 'b': [None, 11], 'c': ['a', 'b']})
+    >>> y = [True, False]
+    >>> X
+                a     b  c
+    0  2020-01-02   NaN  a
+    1  2021-04-01  11.0  b
+    >>> model = make_tabular_pipeline('regressor').fit(X, y)
```

Comment:
I'd probably do
```suggestion
    >>> model = make_tabular_pipeline('classifier').fit(X, y)
```
with a little more date (the current one predicts `[False, False]`)

Reply:
Could be 

```
>>> import pandas as pd
>>> from sklearn.linear_model import LogisticRegression

>>> X = pd.DataFrame(
    {'last_visit': ['2020-01-02', '2021-04-01', '2024-12-05', '2023-08-10'],
     'insulin_prescriptions': [None, 13, 0, 17],
     'fasting_glucose': [35, 140, 44, 137]}
)
>>> y = [False, True, False, True]

>>> model = make_tabular_pipeline(LogisticRegression())
>>> model.fit(X, y)
>>> model.predict(X)
array([False,  True, False,  True])

```